context.go

```diff
@@ -0,0 +1,31 @@
+package log
+
+import (
+	"context"
+)
+
+type contextKey struct{}
+
+// loggerKey is used as key to store logger in context
+var loggerKey contextKey
+
+// ContextWithLogger set *Logger into given context and return
+func ContextWithLogger(ctx context.Context, l *Logger) context.Context {
+	if ctx == nil {
```

Comment:
if `ctx == nil && l == nil` return `Background` directly.
if `l == nil` return ctx
if ctx already has a logger, return directly

Reply:
> if ctx already has a logger, return ctx directly

Should new logger replace the older one?

Reply:
Oh, yes, so we don't need to check this.